qcodes/instrument_drivers/QDevil/QDevil_QDAC.py

```diff
@@ -0,0 +1,938 @@
+# QCoDeS driver for the QDevil QDAC using channels
+# Adapted by QDevil from "qdev\QDac_channels.py" in
+# the instrument drivers package
+# Version 2.0 QDevil 2019-11-19
+
+import time
+import visa
+import logging
+
+from functools import partial
+from qcodes.instrument.channel import InstrumentChannel, ChannelList
+from qcodes.instrument.channel import MultiChannelInstrumentParameter
+from qcodes.instrument.visa import VisaInstrument
+from qcodes.utils import validators as vals
+
+LOG = logging.getLogger(__name__)
+
+MODE_DICT = {0: "V-high / I-high", 1: "V-high / I-low", 2: "V-low / I-low"}
+
+
+class Mode:
+    vhigh_ihigh = 0
+    vhigh_ilow = 1
+    vlow_ilow = 2
+
+
+class Waveform:
+    # Enum-like class defining the built-in waveform types
```

Comment:
Having gone through the code, if have to confess that I feel really challenged by the difficulties that the mode implies.
From the top of my head I would make the Mode class and Enum with values of named tuples. (or a named tuple of enums)

Reply:
Hi Dominik,
Thanks for all the comments. As you probably can see, I am not a trained sw engineer and I grew up with C / C++. Python is "strange and different" to me :-)).
Regarding "the mode" - it was, as you know, your initial comments which made me go that way ;-) But I still think it is the right thing to do. Of course I can make the Mode Class an Enum ( I miss 'real' C enums...), and I can make a named tuple like below for the string representation. But is this what you mean? I don't see the big difference, so probably not...:

class Mode(Enum):
    vhigh_ihigh = 0
    vhigh_ilow = 1
    vlow_ilow = 2
Mode_str = namedtuple('Mode', 'vhigh_ihigh vhigh_ilow vlow_ilow')
mode_str = Mode1('V-high / I-high', 'V-high / I-low', 'V-low / I-low')

Would it be useful if I popped over at NBI instead of discussing online?